jung-visualization/src/test/java/edu/uci/ics/jung/visualization/spatial/rtree/RTreeTest.java

```diff
@@ -227,25 +229,23 @@ private void testAreas(TreeNode rootNode) {
     Rectangle2D rootBounds = rootNode.getBounds();
     if (rootNode instanceof InnerNode) {
       InnerNode innerNode = (InnerNode) rootNode;
-      Assert.assertTrue(closeEnough(rootBounds, Node.union(innerNode.getChildren())));
-    }
 
+      double[] rootDimensions =
+          new double[] {
+            rootBounds.getX(), rootBounds.getY(), rootBounds.getWidth(), rootBounds.getHeight()
+          };
+      Rectangle2D union = Node.union(innerNode.getChildren());
+      double[] unionDimensions =
+          new double[] {union.getX(), union.getY(), union.getWidth(), union.getHeight()};
+      assertWithMessage(
+              "The compared bounds are not close enough: rootBounds: %s, unionBounds: %s",
+              rootBounds, union)
+          .that(rootDimensions)
+          .usingTolerance(1.0E-4)
+          .equals(unionDimensions);
+    }
     for (TreeNode rt : rootNode.getChildren()) {
```

Comment:
I put the 4 doubles {x, y, w, h} into arrays and compared the arrays with a tolerance. It seems like that is more straightforward than building a custom Subject to compare java.awt.Shape instances. Thoughts?

Reply:
It's not entirely clear to me either if it's worth creating a custom `Subject` for `java.awt.Shape` at this stage, so I agree. :)